documents/converters.go

```diff
@@ -271,3 +272,87 @@ func FromP2PPaymentDetails(pdetails []*commonpb.PaymentDetails) ([]*PaymentDetai
 
 	return details, nil
 }
+
+// FromClientCollaboratorAccess converts client collaborator access to CollaboratorsAccess
+func FromClientCollaboratorAccess(racess *documentpb.ReadAccess, waccess *documentpb.WriteAccess) (ca CollaboratorsAccess, err error) {
+	wmap, rmap := make(map[string]struct{}), make(map[string]struct{})
+	var wcs, rcs []string
+	if waccess != nil {
+		for _, c := range waccess.Collaborators {
+			c = strings.ToLower(c)
+			if _, ok := wmap[c]; ok {
+				continue
+			}
+
+			wmap[c] = struct{}{}
+			wcs = append(wcs, c)
+		}
+	}
+
+	if racess != nil {
```

Comment:
same logic as `waccess` so repeated code, Refactor to a function.

Reply:
not the same logic ;)